packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,174 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextField from '../TextField/TextField';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * Determine if a React component is a TextField
+ * @param {React.Node} child - a React component
+ * @return {Boolean} Is this a TextField component?
+ */
+function isTextField(child) {
+  return child != null && child.type === TextField;
+}
+
+/**
+ * The `Autocomplete` component is a parent component that adds autocomplete functionality to a `TextField` component.
+ */
+export class Autocomplete extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+    this.labelId = uniqueId('constrained-list-header_');
+  }
+
+  renderChildren(getInputProps) {
+    // Extend props on the TextField, by passing them through
+    // Downshift's `getInputProps` method
+    return React.Children.map(this.props.children, child => {
+      if (isTextField(child)) {
+        return React.cloneElement(
+          child,
+          getInputProps({
+            fieldClassName: 'ds-c-autocomplete__input',
+            id: this.id,
+            labelClassName: 'ds-u-margin-top--0 ds-c-autocomplete__label'
+          })
+        );
+      }
+
+      return child;
+    });
+  }
+
+  render() {
+    const {
+      clearAriaLabel,
+      clearInputText,
+      items,
+      itemToString,
+      label,
+      onChange
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-clearfix ds-c-autocomplete">
+            {this.renderChildren(getInputProps)}
+
+            {isOpen ? (
```

Comment:
We should handle the case where there are no results. 

From what I'm seeing elsewhere, other implementations hide the autocomplete list container when there aren't any results, so that could be one way we handle this scenario.

A few examples:

- http://ianmcburnie.github.io/mindpatterns/input/autocomplete/
- https://ant.design/components/auto-complete/

Alternatively, if we want to provide feedback to the user that there are no results, we could provide a message in the autocomplete container. This could be an optional prop, which renders the message when present and causes the container to not render when it's not set. 

For example:

```jsx
// This would render a message in the dropdown when there are no results:
<Autocomplete emptyText="No results">...</Autocomplete>

// This would result in no dropdown when there are no results:
<Autocomplete>...</Autocomplete>
```

Here's what things currently look like when the list is empty, which I believe would be confusing to users:

![image](https://user-images.githubusercontent.com/371943/34017117-5d8e3180-e0f2-11e7-8134-4e8b866eaa26.png)

Reply:
Added handling for Loading... and No results.